### PR TITLE
Updating React and ReactDOM dependencies to v15

### PR DESCRIPTION
**Pull request by thathenderson**
I used 15.x.x to try to future-proof the React dependencies; should be good for a little while. (15.0.1 was released while I was doing this, which fixed this cursor issue: https://github.com/facebook/react/issues/6445)

No warnings in the console. I tried `npm test`, but there aren't any tests to run.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,13 +31,13 @@
     "inputmask-core": "^2.1.1"
   },
   "peerDependencies": {
-    "react": "0.14.x"
+    "react": "15.x.x"
```
Comment (szyablitsky) [inline package.json]: i would recommend `"0.14.x || 15.0.x"`

Comment (iamdustan) [inline package.json]: agreed. Either that, or `*`, or remove `peerDeps` entirely.

Comment (thathenderson) [inline package.json]: Thanks, @szyablitsky and @iamdustan! Good points. I like the idea of `0.14.x || 15.0.x`; it doesn't allow major or minor updates, which could introduce breaking changes, and communicates at a glance which versions are supported. Updated!

Comment (iamdustan) [inline package.json]: oh actually I would prefer `15.x.x`

Comment (iamdustan) [inline package.json]: in the `||` column. Minor versions are unlikely to introduce breaking changes and we by default support all future versions. The difference, IMO, will be when we drop support for older versions if newer versions make it difficult.

Comment (thathenderson) [inline package.json]: Makes sense. Updated.

**Comment by iamdustan** [comment]
last nit: could you squash this to a single commit? Then I’ll merge and publish.

**Comment by thathenderson** [comment]
@iamdustan Done!

**Pull request merged**